### DISCUSSION TITLE
HTTPS route with Redirect and HTTP route same address

### DISCUSSION THREAD

**Discussion by cragia** [okd-project/okd#1337] [Q&A]
Hello.
I have a specific question, because I'm using the cert-manager operator in a OKD environment to generate all the certificates.

Now the scenario is this one:
I have an ingress pointing to an address (for example: foo.bar.com), and defined as this:
```yaml
rules:
    - host: foo.bar.com
      http:
        paths:
          - path: /
            pathType: Prefix
```
Then I defined a certificate for the same address. This generates the certificate for the first time with no problem... so after that I have the HTTPS ROUTE (in fact I see the address: https://foo.example.com/) protected with the certificate just created, and inside it I have the rule: `insecureEdgeTerminationPolicy: Redirect`, and that means that if I try to go to http://foo.bar.com, it redirects me to https://foo.bar.com
And that's fine.

Then, after a couple of months, it tries to renew the certificate... so the operator creates a specific ingress, with a specific path, for example:
```yaml
rules:
    - host: foo.bar.com
      http:
        paths:
          - path: /.well-known/...
            pathType: ImplementationSpecific
```
this creates an HTTP ROUTE, with the specific path declared...

The problem seems to be that if I contact the EXACT path specified in the ACME route, it still redirects me to the HTTPS, and than my route matches, and so that request gets passed to my endpoint, and thus failing to contact the service for the certificate generation...

So basically to be able to renew the certificate, I need to delete the certificate, and the generated secret (so that the HTTPS ROUTE gets deleted), and then recreate it, so that now only the HTTP route is present, and so that the check can successfully get completed.

But this means that I always need to do something manually, and that's not ideal... is there a way to fix this problem?

thank you!
